internal/api/handlers/custom.go

```diff
@@ -0,0 +1,23 @@
+package handlers
+
+import (
+	"errors"
+	"net/http"
+)
+
+type HandlerFunc func(w http.ResponseWriter, r *http.Request) error
+
+// CustomHandler is a wrapper that allows us to process and return errors from different handlers.
+func CustomHandler(f HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		err := f(w, r)
+		if err != nil {
+			status := http.StatusInternalServerError
+			var apiError *Error
+			if errors.As(err, &apiError) {
+				status = apiError.HttpStatus()
+			}
+			http.Error(w, err.Error(), status)
```

Comment:
🤔  Hm, I did a quick research and it seems like the result of `http.Error(...` is not a JSON, but a `text/plain`). Is that want you want here?

Reply:
Oooh good catch. I did not realise this. I will use the httpjson.Render method that you pointed out to send error as JSON